test/e2e/specs/editor/various/block-visibility.spec.js

```diff
@@ -0,0 +1,258 @@
+			.getByRole( 'region', { name: 'Available block types' } )
+			.getByRole( 'checkbox', { name: 'Heading' } )
+			.check();
+		await page
+			.getByRole( 'dialog', {
+				name: 'Preferences',
+			} )
+			.getByRole( 'button', { name: 'Close' } )
+			.click();
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+		await page
+			.getByRole( 'region', { name: 'Block Library' } )
+			.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+			.fill( 'Heading' );
+
+		await expect(
+			page
+				.getByRole( 'tabpanel', { name: 'Blocks' } )
+				.getByRole( 'option', { name: 'Heading' } ),
+			'Heading block should be visible'
+		).toBeVisible();
+	} );
+
+	test( 'should hide/show all blocks in a category at once', async ( {
+		page,
+		BlockVisibilityUtils,
+	} ) => {
+		// Hide Media category blocks.
+		await BlockVisibilityUtils.openBlockVisibilityManager();
+		await page
+			.getByRole( 'region', { name: 'Available block types' } )
+			.getByRole( 'checkbox', { name: 'Media', exact: true } )
+			.uncheck();
+		await page
+			.getByRole( 'dialog', {
+				name: 'Preferences',
+			} )
+			.getByRole( 'button', { name: 'Close' } )
+			.click();
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'tabpanel', { name: 'Blocks' } )
+				.getByRole( 'heading', { name: 'Media' } ),
+			'Media category should not be visible'
+		).toBeHidden();
+
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		// Show Media category blocks again.
+		await BlockVisibilityUtils.openBlockVisibilityManager();
+
+		await expect(
+			page
+				.getByRole( 'region', { name: 'Available block types' } )
+				.getByRole( 'checkbox', { name: 'Media', exact: true } ),
+			'Media category should be unchecked'
+		).not.toBeChecked();
+
+		await page
+			.getByRole( 'region', { name: 'Available block types' } )
+			.getByRole( 'checkbox', { name: 'Media', exact: true } )
+			.check();
+		await page
+			.getByRole( 'dialog', {
+				name: 'Preferences',
+			} )
+			.getByRole( 'button', { name: 'Close' } )
+			.click();
+		await page
+			.getByRole( 'button', { name: 'Toggle block inserter' } )
+			.click();
+
+		await expect(
+			page
+				.getByRole( 'tabpanel', { name: 'Blocks' } )
+				.getByRole( 'heading', { name: 'Media' } ),
+			'Media category should be visible'
+		).toBeVisible();
+	} );
+
+	test( 'should search for blocks by keyword', async ( {
+		page,
+		BlockVisibilityUtils,
+	} ) => {
+		await BlockVisibilityUtils.openBlockVisibilityManager();
+		await page
+			.getByRole( 'searchbox', {
+				name: 'Search for a block',
+			} )
+			.fill( 'verse' );
+		const blockCategories = page
+			.getByRole( 'region', {
+				name: 'Available block types',
+			} )
+			.getByRole( 'group' );
+
+		expect(
+			await blockCategories.count(),
+			'Only one category should be visible'
+		).toBe( 1 );
```

Comment:
Nit: Explicit assertions like [`toHaveCount`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-count) should be preferred in similar cases.